core/main/src/bootstrap/extn/load_extn_step.rs

```diff
@@ -15,144 +15,120 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use std::thread;
+
 use ripple_sdk::{
-    api::firebolt::fb_openrpc::OpenRPCParser,
+    api::manifest::extn_manifest::ExtnManifestEntry,
     async_trait::async_trait,
-    extn::{
-        client::extn_sender::ExtnSender,
-        extn_id::ExtnId,
-        ffi::{ffi_channel::load_channel_builder, ffi_jsonrpsee::load_jsonrpsee_methods},
-    },
+    extn::ffi::ffi_channel::load_channel_builder,
     framework::bootstrap::Bootstep,
-    log::{debug, error, info},
+    log::{debug, info, warn},
     utils::error::RippleError,
 };
 
-use crate::state::{
-    bootstrap_state::{BootstrapState, ChannelsState},
-    extn_state::PreLoadedExtnChannel,
-};
-use jsonrpsee::core::server::rpc_module::Methods;
+use crate::state::bootstrap_state::BootstrapState;
+use ripple_sdk::libloading::Library;
+use std::ffi::OsStr;
+
+#[derive(Debug)]
+pub struct LoadedLibrary {
+    pub library: Library,
+    pub entry: ExtnManifestEntry,
+}
+
+impl LoadedLibrary {
+    pub fn new(library: Library, entry: ExtnManifestEntry) -> LoadedLibrary {
+        LoadedLibrary { library, entry }
+    }
+}
 
 /// Actual bootstep which loads the extensions into the ExtnState.
 /// Currently this step loads
 /// 1. Device Channel
 /// 2. Device Extensions
 pub struct LoadExtensionsStep;
 
+impl LoadExtensionsStep {
+    unsafe fn load_extension_library<P: AsRef<OsStr>>(
+        filename: P,
+        entry: ExtnManifestEntry,
+    ) -> Option<LoadedLibrary> {
+        match Library::new(&filename) {
+            Ok(library) => Some(LoadedLibrary::new(library, entry)),
+            Err(err) => {
+                debug!("Extn not found: {:?}", err);
```

Comment:
It would be more helpful to include extension-specific details in the log message, particularly for transitional scenarios where one of the existing extensions is converted to a service while the other remains as an extension (hybrid model).